substrate/frame/glutton/src/lib.rs

```diff
@@ -208,6 +225,58 @@ pub mod pallet {
 		}
 	}
 
+	/// Dummy error type.
+	#[derive(codec::Encode, sp_runtime::RuntimeDebug)]
+	#[cfg_attr(feature = "std", derive(codec::Decode))]
+	pub enum InherentError {
+		/// Fatal error.
+		Fatal,
+	}
+
+	impl frame_support::inherent::IsFatalError for InherentError {
+		fn is_fatal_error(&self) -> bool {
+			matches!(self, InherentError::Fatal)
+		}
+	}
+
+	#[pallet::inherent]
+	impl<T: Config> ProvideInherent for Pallet<T> {
+		type Call = Call<T>;
+		type Error = InherentError;
+
+		const INHERENT_IDENTIFIER: InherentIdentifier = *b"bloated0";
+
+		fn create_inherent(_data: &InherentData) -> Option<Self::Call> {
+			let max_block_length = *T::BlockLength::get().max.get(DispatchClass::Mandatory);
+			let bloat_size = Length::<T>::get().saturating_mul_int(max_block_length) as usize;
+			let amount_trash = bloat_size / VALUE_SIZE;
+			let garbage = TrashData::<T>::iter()
```

Comment:
Why is it taken from the `TrashData`? Can you not just take iterative hashes or so? Otherwise it wont work then that map is empty.  
Then you would also not need to chunk it. Or do you think it would be too slow?

Reply:
The map is populated in genesis or it can also be done with `initialize_pallet`. 
The `TrashData` was previously used to bloat to just bloat the storage proof and since it is random data, it makes perfect sense to use it to fill the garbage parameter for the bloat call.
We also don't want to waste CPU time by doing additional hashing here, but however this will not matter when importing the block as the inherent itself is a no-op.

Reply:
You could also have used the parent hash as seed for a "random generator", but using trash data should also be fine.